torch/distributed/tensor/parallel/fsdp.py

```diff
@@ -220,6 +221,66 @@ def _chunk_tensor(
         )
 
 
+def _chunk_dtensor(
+    tensor: torch.Tensor,
+    rank: int,
+    device_mesh: DeviceMesh,
+) -> DTensor:
+    """
+    Shard a tensor to chunks along the first dimension. The local rank will gets its
+    corresponding chunk as the local tensor to create a DTensor.
+    """
+    parent_mesh = mesh_resources.get_parent_mesh(device_mesh)
+    if parent_mesh is None:
+        raise RuntimeError("No parent device_mesh is found for FSDP device_mesh.")
+    if parent_mesh.ndim != 2:
+        raise RuntimeError(
+            f"Found parent device_mesh of ndim={parent_mesh.ndim},",
+            "but only 2D meshes are currently supported.",
+        )
+
+    # We need to explicitly call .detach() to return a new tensor detached from the current graph.
+    tensor = tensor.clone().detach()
+
+    # if a tensor has not yet sharded by TP
```

Comment:
If we have a model with three layers and we use PairwiseParallel() for TP, the 1st and 2nd layer will be the pair and leave the 3rd layer as it is. In this case, TP would not shard the third layer. 

Similarly, when we have a model with only one layer and we use PairwiseParallel(), it is basically a no-op, since TP doesn't do any sharding to the model.

Reply:
I meant should we add the detail you described in the the comment so that we won't get lost in the future?